cmd/sourced/cmd/status.go

```diff
@@ -2,18 +2,165 @@ package cmd
 
 import (
 	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
 
 	"github.com/src-d/sourced-ce/cmd/sourced/compose"
+	"github.com/src-d/sourced-ce/cmd/sourced/compose/workdir"
 )
 
 type statusCmd struct {
-	Command `name:"status" short-description:"Show the status of all components" long-description:"Show the status of all components"`
+	Command `name:"status" short-description:"Show the list of working directories and the current deployment" long-description:"Show the list of working directories and the current deployment"`
 }
 
-func (c *statusCmd) Execute(args []string) error {
+type statusAllCmd struct {
+	Command `name:"all" short-description:"Show all the available status information" long-description:"Show all the available status information"`
+}
+
+func (c *statusAllCmd) Execute(args []string) error {
+	fmt.Print("List of all working directories:\n")
+
+	err := printWorkdirsCmd()
+	if err != nil {
+		printRed(err.Error())
```

Comment:
is it okay to return 0 exit code in case of an error?

Reply:
I was going to save the error, continue with the other sub commands, and at the end return the original error.
But then the output gets out of order, and we should also consider what happens if 2 sub commands fail, should we append the error messages etc...

So if we are going to return and error in `Execute`, it's much cleaner to just return it here. The possible errors are all unlikely, and if anything fails for a sub command quite probably the other sub commands will also fail.
Done in 3dba9dc

Reply:
awesome. Thanks!